robot/framesystem/parts/framesystem_parts.go

```diff
@@ -5,40 +5,72 @@ import (
 	"fmt"
 	"sort"
 
+	"github.com/golang/geo/r3"
 	"github.com/jedib0t/go-pretty/v6/table"
 	"github.com/pkg/errors"
 
-	"go.viam.com/rdk/config"
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/spatialmath"
 	"go.viam.com/rdk/utils"
 )
 
 // Parts is a slice of *config.FrameSystemPart.
-type Parts []*config.FrameSystemPart
+type Parts []*referenceframe.FrameSystemPart
 
 // String prints out a table of each frame in the system, with columns of name, parent, translation and orientation.
 func (fsp Parts) String() string {
 	t := table.NewWriter()
-	t.AppendHeader(table.Row{"#", "Name", "Parent", "Translation", "Orientation"})
-	t.AppendRow([]interface{}{"0", referenceframe.World, "", "", ""})
+	t.AppendHeader(table.Row{"#", "Name", "Parent", "Translation", "Orientation", "Geometry"})
+	t.AppendRow([]interface{}{"0", referenceframe.World, "", "", "", ""})
 	for i, part := range fsp {
-		tra := part.FrameConfig.Translation
-		ori := &spatialmath.EulerAngles{}
-		if part.FrameConfig.Orientation != nil {
-			ori = part.FrameConfig.Orientation.EulerAngles()
+		pose := part.FrameConfig.Pose()
+		tra := pose.Point()
+		ori := pose.Orientation().EulerAngles()
+		geomString := ""
+		if part.FrameConfig.Geometry() != nil {
```

Comment:
[minor] this section should live in spatialmath as a method on a `GeometryConfig`